Doc/library/ssl.rst

```diff
@@ -563,20 +563,28 @@ Constants
 .. data:: CERT_NONE
 
    Possible value for :attr:`SSLContext.verify_mode`, or the ``cert_reqs``
-   parameter to :func:`wrap_socket`.  In this mode (the default), no
-   certificates will be required from the other side of the socket connection.
-   If a certificate is received from the other end, no attempt to validate it
-   is made.
+   parameter to :func:`wrap_socket`.  Except for :const:`PROTOCOL_TLS_CLIENT`,
+   it is the default mode.  With client-side sockets, just about any
```

Comment:
For a moment I feared that you have discovered a security bug.

Thankfully the code is robust and doesn't allow you to use ``PROTOCOL_TLS_CLIENT`` with default arguments of ``wrap_socket`` and ``SSLSocket``. Both ``ssl.wrap_socket(ssl_version=ssl.PROTOCOL_TLS_CLIENT)`` and ``ssl.SSLSocket(ssl_version=ssl.PROTOCOL_TLS_CLIENT)`` will both fail with error ``Cannot set verify_mode to CERT_NONE when check_hostname is enabled.``.

So the insecure default doesn't work here. I consider it a feature. :) I'm going to deprecate ``ssl.wrap_socket`` and ``ssl.SSLSocket()`` default constructor because they have more fundamental issues.

Reply:
Okay, now I realize that the PROTOCOL_TLS_CLIENT mode is newer than the old _wrap_socket_ function. It was not clear that _wrap_socket_ doesn’t support this new mode, but that is separate to what you are trying to do here.